app/src/pages/index.js

```diff
@@ -5,16 +5,40 @@ import SEO from "../components/seo"
 import BookCard from "../components/bookcard"
 
 export default ({data}) => {
+  let max = data.allBooksJson.edges.length;
+  let init = 4;
+  let end = 8;
+  function hide(){
+    for(var i=init; i<document.getElementsByClassName("book").length; i++){
+      document.getElementsByClassName("book")[i].hidden=true;
+    }
+  }
+  window.document.onscroll = () => myFunction();
+  function myFunction() {
+    if (document.documentElement.clientHeight+document.documentElement.scrollTop===document.documentElement.scrollHeight && end<=max) {
+      for(var i=init; i<end; i++){
+        console.log(data.allBooksJson.edges[i].node);
```

Comment:
This line is actually fetching just enough books at a time.

I just didn't know how to exactly modify the current book fetching process(which returns all without any condition).